### PR TITLE
Add team module tables, change plan primary key to numeric

### LINKED ISSUE #20: Create table creation scripts for Team module tables
Create table creation scripts for Team module tables

- [x] organization
- [x] practitioner
- [x] practitioner_role
- [x] organization_location
- [ ] ~~organization_plan~~

### LINKED ISSUE #21: Change Plan table primary key to numerical key

### PR DESCRIPTION
**Pull request by githengi**
Resolves #20 and Resolves #21

### CODE REVIEW

File: assets/migrations/scripts/20190829140331_change_plan_id_to_numeric.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+DROP INDEX IF EXISTS core.plan_id_index;
+ALTER TABLE core.plan_metadata RENAME COLUMN plan_id TO identifier;
+ALTER TABLE core.plan_metadata ADD COLUMN  plan_id bigserial;
+UPDATE core.plan_metadata SET plan_id = (SELECT id FROM core.plan p where p.identifier=core.plan_metadata.identifier);
+CREATE INDEX plan_id_index ON core.plan_metadata (plan_id);
+
+
+-- //@UNDO
+-- SQL to undo the change goes here.
+ALTER TABLE core.plan DROP CONSTRAINT IF EXISTS plan_pkey;
```
Comment (moshthepitt) [inline assets/migrations/scripts/20190829140331_change_plan_id_to_numeric.sql]: Should this be adding `plan_pkey` the constraint back, when undoing?

File: assets/migrations/scripts/20190829154642_create_organization_table.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+-- // create_organization_table
+-- Migration SQL that makes the change goes here.
+
+CREATE TABLE team.organization
+(
+	id bigserial NOT NULL,	
+    identifier character varying NOT NULL,
+    active boolean default false,
+    name character varying NOT NULL,
+    type jsonb ,
```
Comment (moshthepitt) [inline assets/migrations/scripts/20190829154642_create_organization_table.sql]: Curious, what is being stored in this field?  Also, could we remove the space before the comma?

File: assets/migrations/scripts/20190829154642_create_organization_table.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+--    See the License for the specific language governing permissions and
+--    limitations under the License.
+--
+
+-- // create_organization_table
+-- Migration SQL that makes the change goes here.
+
+CREATE TABLE team.organization
+(
+	id bigserial NOT NULL,
```
Comment (moshthepitt) [inline assets/migrations/scripts/20190829154642_create_organization_table.sql]: Could we remove the whitespace here?

File: assets/migrations/scripts/20190829140331_change_plan_id_to_numeric.sql (hunk 1 above)
Comment (githengi) [reply assets/migrations/scripts/20190829140331_change_plan_id_to_numeric.sql]: The primary key is added back after changing the field on this line
`ALTER TABLE core.plan ADD PRIMARY KEY (id);`

File: assets/migrations/scripts/20190829154642_create_organization_table.sql (hunk 2 above)
Comment (githengi) [reply assets/migrations/scripts/20190829154642_create_organization_table.sql]: This is the organization terminology code system. I suspect we may not need in openSRP or Reveal however its important when exchanging data via FHIR.

The data to be stored is in the below format 
```
    "coding": [
      {
        "system": "http://terminology.hl7.org/CodeSystem/organization-type",
        "code": "team",
        "display": "Team"
      }
    ]
```
This is per the spec https://smartregister.atlassian.net/wiki/spaces/REV/pages/1159921677/Team+Structure+and+User+Assignment

File: assets/migrations/scripts/20190829154642_create_organization_table.sql (hunk 3 above)
Comment (githengi) [reply assets/migrations/scripts/20190829154642_create_organization_table.sql]: The white-space is for readability

**Pull request merged**